src/PrestaShopBundle/EventListener/Admin/Context/ShopContextListener.php

```diff
@@ -183,4 +198,29 @@ private function redirectShopContext(RequestEvent $requestEvent): ?RedirectRespo
             ['setShopContext', 'conf']
         ));
     }
+
+    /**
+     * @throws ReflectionException
+     */
+    private function verifyAllShopAttribute(Request $request): ?ShopConstraint
+    {
+        try {
+            $routeInfo = $this->router->match($request->getPathInfo());
+            $controller = $routeInfo['_controller'];
+            [$className, $methodName] = explode('::', $controller);
+
+            $reflectionClass = new ReflectionClass($className);
+            $classAttributes = $reflectionClass->getAttributes(AllShopContext::class);
+            $methodAttributes = $reflectionClass->getMethod($methodName)->getAttributes(AllShopContext::class);
```

Comment:
I'm not sure about this, but in case `$classAttributes` is not empty isn't this enough? You return the shop constraint right away no? And only check the method attributes when it's empty I guess (again in a spirit of optimisation, although this one would probably be negligeable 😅)

Reply:
For me the class may not have an attribute. If I have an attribute on a method of my class, and not directly the class, then the attributes of the class will be empty while those of my methods will not be empty. Much like IsGranted attributes, where you can secure certain routes, the controller itself is not.

Reply:
Ah alright let's go with this then! 😉